### PR TITLE
Update squashing docs

### PR DESCRIPTION
**Pull request by JoshSmith**
This updates the squashing docs to reference any editor, not just Sublime. Also breaks them out into their own separate area.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,43 +35,4 @@ Before We Can Merge
 
 If you've had a pull request reviewed and accepted, congratulations! Before we can merge your changes, we'll need you to rebase off `origin/develop` and squash your commits into one. This will give us a cleaner git history.
```
Comment (noslouch) [inline CONTRIBUTING.md]: Potentially change to "rebase onto" to match how it's referred to elsewhere (like in the universe)

File: docs/SQUASHING.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# How do I "squash" my commits?
+
+Before you start, [make sure your text editor is associated with Git](https://help.github.com/articles/associating-text-editors-with-git/) to not close without a response, otherwise you'll see some weird behavior.
+
+On your command line, run:
```
Comment (noslouch) [inline docs/SQUASHING.md]: The first step should probably be git fetch --all so their refs are updated

**Comment by noslouch** [comment]
If this is really a first timer's guide, should we also walk through working off a fork and setting up two different remotes? One that points to code-corps and another that points to their own repo?

**Comment by JoshSmith** [comment]
@noslouch that makes sense. Tackle that in a separate issue maybe?

**Comment by noslouch** [comment]
yup, another issue sounds fine for this kind of thing

**Comment by JoshSmith** [comment]
@noslouch do you want to open that?

**Comment by noslouch** [comment]
@JoshSmith #326

**Pull request merged**